lib/_stream_writable.js

```diff
@@ -362,27 +388,34 @@ function onwriteDrain(stream, state) {
 function clearBuffer(stream, state) {
   state.bufferProcessing = true;
   var entry = state.bufferedRequest;
+  var bufferedRequests = state.bufferedRequests;
+
+  state.bufferedRequests = 0;
 
   if (stream._writev && entry && entry.next) {
     // Fast case, write everything using _writev()
-    var buffer = [];
-    var cbs = [];
+    var buffer = new Array(bufferedRequests);
```

Comment:
@lucamaraschi Might not be. But if we had to modify one of the arrays, then the other array also has to updated.